### PR TITLE
[iOS] Clip Border content based on StrokeShape property

### LINKED ISSUE #7475: Border does not clip when stroke shape is provided in iOS and maccatalyst
### Description

Border does not clip and the child gets overlapped.

`<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="TestApp.MainPage">

    <Grid>
        <Border HeightRequest="200" WidthRequest="200" StrokeThickness="5" Stroke="Red">
            <Border.StrokeShape>
                <RoundRectangle CornerRadius="100" />
            </Border.StrokeShape>

            <Image Background="yellow"
                Source="dotnet_bot.png"
                HorizontalOptions="Center">
            </Image>
        </Border>
    </Grid>
</ContentPage>`

image

### Steps to Reproduce

Run the attached code snippet in a .NET MAUI application

### Version with bug

.NET MAUI GA

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, macOS

### Affected platform versions

iOS15

### Did you find any workaround?

No

### Relevant log output

_No response_

### LINKED ISSUE #7493: The indicatorView render outside of border
### Description

I defined a CarouselView and Indicator view. They are both the child of VerticalStackLayout. The VerticalStackLayout is the Child of Border. But the Indicator View remains outside of border on iOS. 

### Steps to Reproduce

Run the repro project. Scroll down and see the indicatorview at the bottom of the screen outside of border.
[MauiRepro.zip](https://github.com/dotnet/maui/files/8778151/MauiRepro.zip)

### Version with bug

6.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 15

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Currently is not implemented:

Captura de Pantalla 2022-05-19 a las 13 57 26

We implemented it on Android and Windows: https://github.com/dotnet/maui/pull/5789

This PR include changes to clip the Border Content based on StrokeShape property:
clip-border-ios

Fixes https://github.com/dotnet/maui/issues/7475
Fixes #7493
Fixes https://github.com/dotnet/maui/issues/8381

### CODE REVIEW

File: src/Core/src/Platform/iOS/ContentView.cs [hunk 1]
```diff
@@ -37,7 +47,90 @@ public override void SetNeedsLayout()
 			Superview?.SetNeedsLayout();
 		}
 
+		public new void Dispose()
+		{
+			DisposeClip();
+
+			base.Dispose();
+		}
```
Comment (mattleibow) [inline src/Core/src/Platform/iOS/ContentView.cs:55]: Is the UIView not disposable? Should we override?

Also, @PureWeen said things about never disposing, so I wonder what this means. Should we let the GC take control?

**Review by mattleibow — CHANGES_REQUESTED** [review]
@PureWeen the code looks valid, but the dispose scares me.

File: src/Core/src/Platform/iOS/ContentView.cs (hunk 1 above)
Comment (rmarinho) [reply src/Core/src/Platform/iOS/ContentView.cs:55]: @jsuarezruiz  do you need this to fix the issue?

**Review by rmarinho — CHANGES_REQUESTED** [review]
Needs rebase, also can we remove the dispose?

**Comment by rmarinho** [comment]
@jsuarezruiz @mattleibow i removed the dispose When the ContentView goes away , the layer should bealso grabbed and dispoed, no need to set to null, right @PureWeen  ?

**Review by rmarinho — APPROVED** [review]

**Review by PureWeen — APPROVED** [review]

**Pull request merged**